matcher/matcher.py

```diff
@@ -0,0 +1,109 @@
+import jmespath
+import json
+import re
+
+
+
+
+def apply_operator(j_val, op, val):
+
+	if j_val == None:
+		return False
+
+	all_num_ops = ['==', '!=', '<', '<=', '>', '>=']
+	all_str_ops = ['eq', 'ne', 'startswith', 'endswith', 'matches']
+	all_list_ops = ['in', 'contains']
+	
+	# all numeric operations
+	if op in all_num_ops:
+		assert isinstance(j_val, (int, float)), 'value is not numeric'
+		assert isinstance(val, (int, float)), 'value is not numeric'
+		
+		if op == '==':
+			return j_val == val
+		
+		if op == '!=':
+			return j_val != val
+		
+		if op == '>':
+			return j_val > val
+
+		if op == '>=':
+			return j_val >= val
+		
+		if op == '<':
+			return j_val < val
+
+		if op == '<=':
+			return j_val <= val
+
+
+	# string operations
+	if op in all_str_ops:
+		assert isinstance(j_val,(str,unicode)), 'value is not str'
+		assert isinstance(val,(str,unicode)), 'value is not str'
+		
+		if op == 'eq':
+			return j_val == val
+
+		if op == 'ne':
+			return j_val != val
+
+		if op == 'startswith':
+			return j_val.startswith(val)
+	
+		if op == 'endswith':
+			return j_val.endswith(val)
+
+		if op == 'matches':
+			if len(re.findall(val,j_val)) > 0:
+				return True
+			else: 
+				return False
+
+	# list operators
+	if op in all_list_ops:
+
+		if op == 'in':
+			assert isinstance(j_val,(str,unicode)), 'value is not str'
+			assert isinstance(val,(list,)), 'value is not list'
+			return j_val in val
+
+		if op == 'contains':
+			assert isinstance(j_val,(list,)), 'value is not str'
+			assert isinstance(val,(str,unicode)), 'value is not list'
+			return val in j_val
+
+
+
+def match_specs(match_expression, ironic_config):
```

Comment:
This code doesn't really care of the data comes from ironic or somewhere else, so it's probably misleading to call this variable `ironic_config`.  Maybe just `data`?